### PR TITLE
Intial draft for maintainer doc

### PR DESCRIPTION
**Pull request by Sherwin-14**
I've made this doc as per my understanding. Open to discussing what more/less could be done here.

----
📚 Documentation preview 📚: https://earthaccess--608.org.readthedocs.build/en/608/

### CODE REVIEW

File: docs/maintainers-guide.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+```environment-mindeps.yaml``` - The test-environment file lists required and test dependencies for a testing environment.
+
+# Continuous Documentation
+
+The Read the Docs service is used to preview documentation changes made in pull requests. This service uses a configuration file, ```.readthedocs.yaml```.
+
+# Making a Release
+
+To automate releases, we employ a script called ```publish.sh``` in the scripts folder. This script utilizes Poetry for managing and publishing the package to PyPI.
```
Comment (mfisher87) [inline docs/maintainers-guide.md:58]: Let's re-use our existing releasing page `contributing/releasing.md` instead :)

File: docs/maintainers-guide.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
+# Maintainers Guide
+
+This page offers guidance to project maintainers regarding our setup procedures, release processes, package creation, and other related tasks.
+
+# Maintainer Onboarding and Best Practices
```
Comment (mfisher87) [inline docs/maintainers-guide.md:5]: Let's reserve header level 1 for the title of the document, and add one extra header level to all the other sub-headers.

File: docs/maintainers-guide.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+## Becoming a Maintainer or Triager
+
+If you are interested in becoming a maintainer, you can join our community. We expect you to be actively involved.
+
+Also, if you're interested in helping managing issues with labels and interacting with incoming requests, you can have a "triager" role!
```
Comment (mfisher87) [inline docs/maintainers-guide.md:11]:
```suggestion
Also, if you're interested in helping managing issues with labels and interacting with incoming requests, you can have a "triager" role!

To get permissions, please start by participating on GitHub by answering questions, reviewing PRs, or contributing code or documentation. Once you're feeling comfortable, you can ask any of our maintainers for permissions by `@`ing them on GitHub.
```

File: docs/maintainers-guide.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
+# Maintainers Guide
+
+This page offers guidance to project maintainers regarding our setup procedures, release processes, package creation, and other related tasks.
+
+# Maintainer Onboarding and Best Practices
+
+## Becoming a Maintainer or Triager
+
+If you are interested in becoming a maintainer, you can join our community. We expect you to be actively involved.
```
Comment (mfisher87) [inline docs/maintainers-guide.md:9]:
```suggestion
If you are interested in becoming a maintainer, you can join our community. Maintainers have several important responsibilities, so please read on to understand the role.
```

I feel the "actively involved" part is more clearly explained in the next section :)

File: docs/maintainers-guide.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+    3a. Review contributions thoroughly.
+
+    3b. Provide constructive feedback.
+
+    3c. Communicate clearly and respectfully.
+
+    3d. Merge contributions promptly.
+
+4. As a maintainer, you will be releasing different versions. More on this in [here](#making-a-release).
```
Comment (mfisher87) [inline docs/maintainers-guide.md:36]: The above two sections are exactly what we need! Thank you :star_struck:

File: docs/maintainers-guide.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+# Branches
+
+main: This is the main branch, which is consistently tested and prepared for release as a new version. Avoid pushing changes directly to this branch. Instead, create a new branch and submit a pull request for any modifications.
+
+gh-pages: This branch hosts HTML docs, updated by GitHub Actions. Don't commit directly. Make changes in a separate branch, then submit a pull request. GitHub Actions will automatically update docs on gh-pages when merged.
```
Comment (mfisher87) [inline docs/maintainers-guide.md:42]:
```suggestion
```

Our `gh-pages` branch is static and only redirects to ReadTheDocs, so we can remove this section.

File: docs/maintainers-guide.md [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+
+# Continuous Integration
+
+The GitHub Actions CI services handle the project's building, testing, and management across Linux, macOS, and Windows platforms. The CI file manages the continuous integration processes, and list out their respective dependencies. Here we briefly summarize the functions of the files in CI.
```
Comment (mfisher87) [inline docs/maintainers-guide.md:46]:
```suggestion
The GitHub Actions CI services handle the project's building, testing, and management across Linux, macOS, and Windows platforms. The CI workflow manages the continuous integration processes, and list out their respective dependencies. Here we briefly summarize the functions of the files in CI.
```

TODO: Link to the CI workflow YAML file being referenced here

File: docs/maintainers-guide.md [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+
+4. As a maintainer, you will be releasing different versions. More on this in [here](#making-a-release).
+
+# Branches
+
+main: This is the main branch, which is consistently tested and prepared for release as a new version. Avoid pushing changes directly to this branch. Instead, create a new branch and submit a pull request for any modifications.
+
+gh-pages: This branch hosts HTML docs, updated by GitHub Actions. Don't commit directly. Make changes in a separate branch, then submit a pull request. GitHub Actions will automatically update docs on gh-pages when merged.
+
+# Continuous Integration
```
Comment (mfisher87) [inline docs/maintainers-guide.md:44]: We could call this "Continuous Integration & Delivery". We have quite a few workflows going on, and I don't think we need to go in depth on all of them right now, we can always build that out more later. But I do think it's important to call out that delivery / releasing is part of our workflows :)

File: docs/maintainers-guide.md [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+
+The GitHub Actions CI services handle the project's building, testing, and management across Linux, macOS, and Windows platforms. The CI file manages the continuous integration processes, and list out their respective dependencies. Here we briefly summarize the functions of the files in CI.
+
+```environment-dev.yml ```  - The environment configuration file specifies dependencies for Python development.
+
+```environment-mindeps.yaml``` - The test-environment file lists required and test dependencies for a testing environment.
```
Comment (mfisher87) [inline docs/maintainers-guide.md:50]:
```suggestion
```environment-mindeps.yaml``` - Specifies an environment with the minimum supported dependency versions to detect backwards-incompatible code.
```

File: docs/maintainers-guide.md [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+
+# Continuous Documentation
+
+The Read the Docs service is used to preview documentation changes made in pull requests. This service uses a configuration file, ```.readthedocs.yaml```.
```
Comment (mfisher87) [inline docs/maintainers-guide.md:54]:
```suggestion
[ReadTheDocs](https://readthedocs.org/projects/earthaccess/) is used to generate and host [our documentation website](https://earthaccess.readthedocs.io/) as well as preview documentation changes made in pull requests. This service uses a configuration file, ```.readthedocs.yaml```.
```

File: docs/maintainers-guide.md [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+
+main: This is the main branch, which is consistently tested and prepared for release as a new version. Avoid pushing changes directly to this branch. Instead, create a new branch and submit a pull request for any modifications.
+
+gh-pages: This branch hosts HTML docs, updated by GitHub Actions. Don't commit directly. Make changes in a separate branch, then submit a pull request. GitHub Actions will automatically update docs on gh-pages when merged.
+
+# Continuous Integration
+
+The GitHub Actions CI services handle the project's building, testing, and management across Linux, macOS, and Windows platforms. The CI file manages the continuous integration processes, and list out their respective dependencies. Here we briefly summarize the functions of the files in CI.
+
+```environment-dev.yml ```  - The environment configuration file specifies dependencies for Python development.
```
Comment (mfisher87) [inline docs/maintainers-guide.md:48]:
```suggestion
`environment-dev.yml`  - The environment configuration file specifies dependencies for Python development.
```

Please change any inline code to single backticks throughout :bow:

**Review by mfisher87 — CHANGES_REQUESTED** [review]
This looks great! Thanks so much for your work. A couple nitpicky suggestions :)

File: docs/maintainers-guide.md (hunk 5 above)
Comment (Sherwin-14) [reply docs/maintainers-guide.md:36]: Could you please be more specific about this? I'm kind of confused here.

Comment (mfisher87) [reply docs/maintainers-guide.md:36]: Sorry for being unclear. I was saying that I think the "Maintainer Responsibilities and Expectations" & " Maintainer Processes Beyond Regular Contributing" sections look great :)

**Comment by mfisher87** [comment]
pre-commit.ci autofix

**Review by mfisher87 — COMMENTED** [review]
Looks great, just one last thing: There's currently no link to get to this page. What do you think of including it in the "Work with us" part of the side nav (`mkdocs.yml`)?

I was thinking we should also move the file into the `doc/contributing` directory. Do you agree?

**Review by mfisher87 — APPROVED** [review]

**Comment by mfisher87** [comment]
@Sherwin-14 LGTM! Let me know if you're ready to merge or if you'd like some time to rebase beforehand.

**Comment by Sherwin-14** [comment]
@mfisher87 I am ready to merge, you can decide as per your convenience :)

**Comment by mfisher87** [comment]
Thank you for your contribution :)

**Pull request merged**